src/launch/lib/automation/terragrunt/functions.py

```diff
@@ -28,13 +37,33 @@ def terragrunt_init(run_all=True):
         subprocess_args = ["terragrunt", "init", "--terragrunt-non-interactive"]
 
     try:
-        subprocess.run(subprocess_args, check=True)
+        if dry_run:
+            click.secho(
+                f"[DRYRUN] Would have ran subprocess: {subprocess_args=}",
+                fg="yellow",
+            )
+        else:
+            subprocess.run(subprocess_args, check=True)
     except subprocess.CalledProcessError as e:
         raise RuntimeError(f"An error occurred: {str(e)}")
 
 
-def terragrunt_plan(file=None, run_all=True):
-    logger.info("Running terragrunt plan")
+def terragrunt_plan(file=None, run_all=True, dry_run=True) -> None:
+    """
+    Runs terragrunt plan subprocess in the current directory.
+
+    Args:
+        file (str, optional): The file to pass to terragrunt. Defaults to None.
```

Comment:
Can we clarify the usage of `file` by adding type hints and clarifying in this docstring? It looks like this is used for the plan output which isn't really reflected here. Might also be worthwhile to add verbiage indicating that plan files are not saved when using the `run_all` argument.

The other two TG commands in this file (apply, destroy) use this argument as a way for us to pass their variables in, I would like to make sure that distinction is clear between these functions in their documentation. I would also be amenable to changing the argument name for this to be plan_output_file or similar to help distinguish them.